make/ide/eclipse/CreateWorkspace.gmk

```diff
@@ -0,0 +1,340 @@
+#
+# This code is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+# version 2 for more details (a copy is included in the LICENSE file that
+# accompanied this code).
+#
+# You should have received a copy of the GNU General Public License version
+# 2 along with this work; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+# or visit www.oracle.com if you need additional information or have any
+# questions.
+#
+
+default: all
+
+include $(SPEC)
+include MakeBase.gmk
+include Utils.gmk
+include Modules.gmk
+
+IDE_OUTPUTDIR := $(OUTPUTDIR)/ide/eclipse
+
+CLASSPATH_FILE := $(IDE_OUTPUTDIR)/.classpath
+WORKSPACE_FILE := $(IDE_OUTPUTDIR)/.project
+NATIVE_FILE := $(IDE_OUTPUTDIR)/.cproject
+ANT_BUILDFILE := $(IDE_OUTPUTDIR)/build.xml
+
+# FixPath already removes $(FIXPATH) automatically, but if we rely on FixPath
+# to do it for us dir won't be able to handle the converted path (if on Windows).
+# Reversing the order causes dir to mangle the preceeding $(FIXPATH) which breaks
+# the ability of FixPath to remove it, so it has to be manually removed here.
+# This currently does not support Visual C++
+COMPILER_PATH := $(call FixPath, $(dir $(patsubst $(FIXPATH), , $(CC))))
+
+# Sidestep annoying issue of WSL path conversion not working properly with true
+# Unix paths. Eclipse can't run the WSL make by itself anyway, even if we could
+# properly convert the path.
+ifeq ($(call isBuildOsEnv, windows.wsl1 windows.wsl2), true)
+  MAKE_EXE := wsl.exe $(MAKE)
+else
+  MAKE_EXE := $(call FixPath, $(MAKE))
+endif
+
+# Random number generation, with the amount of specified digits
+# No need to check if the parameter passed is a valid number, the shell will
+# be more than happy to complain if it isn't ;)
+RandomDigits = \
+    $(shell od -An -N8 -d /dev/urandom | sed -e 's| ||g' -e 's|\(.\{$(strip $1)\}\).*|\1|')
+
+# Modified version of utility from JdkNativeCompilation.gmk
+# The full path could be omitted, but it was chosen not to so wildcard
+# can work properly for existing directories. The pathsubst below that
+# this necessitates is a relatively small price to pay.
+FindModuleNativeDirs = \
+  $(call uniq, $(wildcard \
+      $(TOPDIR)/src/$(strip $1)/$(OPENJDK_TARGET_OS)/native \
+      $(TOPDIR)/src/$(strip $1)/$(OPENJDK_TARGET_OS_TYPE)/native \
+      $(TOPDIR)/src/$(strip $1)/share/native))
+
+JAVA_DIRS := $(strip $(foreach module, $(call FindAllModules), \
+    $(patsubst $(TOPDIR)/%,%,$(filter-out $(OUTPUTDIR)%, \
+    $(call FindModuleSrcDirs, $(module))))))
+
+NATIVE_DIRS := $(strip $(foreach module, $(call FindAllModules), \
+    $(patsubst $(TOPDIR)/%,%,$(call FindModuleNativeDirs, $(module)))))
+
+EmitEmptyLine = $(file >> $(1),)
```

Comment:
Note that using the file macro requires GNU make >=4.0. You probably want to document this requirement and add a fast fail if it's not fulfilled.

Reply:
I think it would be better if I looked for something that the make system already has and used that instead of something that's tied to a specific version and GNU specific now that I'm revisiting this, thanks for the catch